server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3433,12 +3433,23 @@ public void activateWithPrimaryContext(final ReplicationTracker.PrimaryContext p
             + "] does not contain relocation target ["
             + routingEntry()
             + "]";
-        assert getLocalCheckpoint() == primaryContext.getCheckpointStates().get(routingEntry().allocationId().getId()).getLocalCheckpoint()
-            || indexSettings().getTranslogDurability() == Durability.ASYNC : "local checkpoint ["
-                + getLocalCheckpoint()
-                + "] does not match checkpoint from primary context ["
-                + primaryContext
-                + "]";
+        if (isRemoteStoreEnabled()) {
```

Comment:
Lets add in some tests cases to confirm this behavior

Reply:
There already a test exists: `org.opensearch.remotestore.RemoteIndexPrimaryRelocationIT.testPrimaryRelocationWhileIndexing` but it is flaky without this change

Reply:
From the annotation below this codepath isn't hit, is the coverage tool incorrect?

Reply:
The problem is: coverage tool does not consider code covered by integ tests

https://github.com/opensearch-project/OpenSearch/blob/4323af1973c6c19b7ab35754f3ff1d6a5437dce9/gradle/code-coverage.gradle#L37-L44

Reply:
Let me add unit test around this change.

Reply:
Thanks @sachinpkale !